test/e2e/kubernetes/kubernetes_test.go

```diff
@@ -556,10 +556,18 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
 				r := rand.New(rand.NewSource(time.Now().UnixNano()))
 				serviceName := "ingress-nginx"
 				deploymentName := fmt.Sprintf("ingress-nginx-%s-%v", cfg.Name, r.Intn(99999))
+				d, _ := deployment.Get(deploymentName, "default")
+				if d != nil {
+					_ = d.Delete(deleteResourceRetries)
```

Comment:
shouldn't we catch the error here if the deployment exists but we aren't able to delete it?

Reply:
I'm trying to be incremental and permissive here, in the event there's a race condition between the delete operation in a prior run (soak test scenario), say the deployment exists but is in a state of being deleted.

But sure, but perhaps throwing an error here is the more incremental approach in terms of figuring out which parts of the resource scaffolding is brittle over successive E2E runs against a long-running cluster.